### ISSUE TITLE
Parse errors in clean compiling code

### ISSUE THREAD

**Issue by daFreeMan** [rubberduck-vba/Rubberduck#3130]
For the last couple of days, I've been getting parse errors in clean compiling code in Excel2010, RD 2.0.13, Win7.

Looks like the error is here:

```
2017-07-14 09:09:35.0032;ERROR-2.0.13.32288;Rubberduck.Parsing.VBA.ParseCoordinator;Unexpected exception thrown in parsing run. (thread 93).;System.AggregateException: One or more errors occurred. ---> System.Runtime.InteropServices.COMException: Error HRESULT E_FAIL has been returned from a call to a COM component.
   at Microsoft.Vbe.Interop._VBComponent.get_Type()
   at Rubberduck.VBEditor.SafeComWrappers.VBA.VBComponent.get_Type() in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.VBEEditor\SafeComWrappers\VBA\VBComponent.cs:line 20
   at Rubberduck.VBEditor.QualifiedModuleName..ctor(IVBComponent component) in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.VBEEditor\QualifiedModuleName.cs:line 47
   at Rubberduck.Parsing.VBA.RubberduckParserState.SetModuleState(IVBComponent component, ParserState state, SyntaxErrorException parserError, Boolean evaluateOverallState) in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.Parsing\VBA\RubberduckParserState.cs:line 367
   at Rubberduck.Parsing.VBA.ParseCoordinator.<>c__DisplayClass32_0.<SetModuleStates>b__0(IVBComponent component) in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.Parsing\VBA\ParseCoordinator.cs:line 264
   at System.Threading.Tasks.Parallel.<>c__DisplayClass31_0`2.<ForEachWorker>b__0(Int32 i)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass17_0`1.<ForWorker>b__1()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.InnerInvokeWithArg(Task childTask)
   at System.Threading.Tasks.Task.<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0(Object )
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at System.Threading.Tasks.Parallel.ForWorker[TLocal](Int32 fromInclusive, Int32 toExclusive, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Func`4 bodyWithLocal, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal](IList`1 list, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEachWorker[TSource,TLocal](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Action`3 bodyWithStateAndIndex, Func`4 bodyWithStateAndLocal, Func`5 bodyWithEverything, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.ForEach[TSource](IEnumerable`1 source, ParallelOptions parallelOptions, Action`1 body)
   at Rubberduck.Parsing.VBA.ParseCoordinator.SetModuleStates(ICollection`1 components, ParserState parserState, CancellationToken token) in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.Parsing\VBA\ParseCoordinator.cs:line 264
   at Rubberduck.Parsing.VBA.ParseCoordinator.ResolveAllReferences(ICollection`1 toResolve, CancellationToken token) in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.Parsing\VBA\ParseCoordinator.cs:line 516
   at Rubberduck.Parsing.VBA.ParseCoordinator.ExecuteCommonParseActivities(ICollection`1 toParse, CancellationToken token) in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.Parsing\VBA\ParseCoordinator.cs:line 243
   at Rubberduck.Parsing.VBA.ParseCoordinator.ParseAllInternal(Object requestor, CancellationToken token) in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.Parsing\VBA\ParseCoordinator.cs:line 749
   at Rubberduck.Parsing.VBA.ParseCoordinator.ParseAll(Object requestor, CancellationToken token) in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.Parsing\VBA\ParseCoordinator.cs:line 682
---> (Inner Exception #0) System.Runtime.InteropServices.COMException (0x80004005): Error HRESULT E_FAIL has been returned from a call to a COM component.
   at Microsoft.Vbe.Interop._VBComponent.get_Type()
   at Rubberduck.VBEditor.SafeComWrappers.VBA.VBComponent.get_Type() in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.VBEEditor\SafeComWrappers\VBA\VBComponent.cs:line 20
   at Rubberduck.VBEditor.QualifiedModuleName..ctor(IVBComponent component) in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.VBEEditor\QualifiedModuleName.cs:line 47
   at Rubberduck.Parsing.VBA.RubberduckParserState.SetModuleState(IVBComponent component, ParserState state, SyntaxErrorException parserError, Boolean evaluateOverallState) in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.Parsing\VBA\RubberduckParserState.cs:line 367
   at Rubberduck.Parsing.VBA.ParseCoordinator.<>c__DisplayClass32_0.<SetModuleStates>b__0(IVBComponent component) in C:\Users\Mathieu\Documents\GitHub\Rubberduck (main)\Rubberduck\Rubberduck.Parsing\VBA\ParseCoordinator.cs:line 264
   at System.Threading.Tasks.Parallel.<>c__DisplayClass31_0`2.<ForEachWorker>b__0(Int32 i)
   at System.Threading.Tasks.Parallel.<>c__DisplayClass17_0`1.<ForWorker>b__1()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.InnerInvokeWithArg(Task childTask)
   at System.Threading.Tasks.Task.<>c__DisplayClass176_0.<ExecuteSelfReplicating>b__0(Object )<---

2017-07-14 09:09:35.1122;DEBUG-2.0.13.32288;Rubberduck.Parsing.VBA.RubberduckParserState;RubberduckParserState raised StateChanged (Error);
2017-07-14 09:09:35.2282;DEBUG-2.0.13.32288;Rubberduck.Parsing.VBA.ParseCoordinator;Parsing run finished after 33s. (thread 93).;

```

Full RD log attached in case something else needs to be looked at.

[RubberduckLog.txt](https://github.com/rubberduck-vba/Rubberduck/files/1148491/RubberduckLog.txt)

**Comment by retailcoder** [comment]
Is it possible that a module was removed? At run-time?

**Comment by MDoerner** [comment]
@daFreeMan Do you still get this error or can we close this?

**Comment by daFreeMan** [comment]
This has been resolved in [next] build 2.1.2003
